Source/events.filtering/Internal/EventFilterProcessor.ts

```diff
@@ -68,8 +68,8 @@ export class EventFilterProcessor extends FilterEventProcessor<FilterRegistratio
         return response;
     }
 
-    protected filter(event: any, context: EventContext): FilterResponse {
-        const shouldInclude = this._callback(event, context);
+    protected async filter(event: any, context: EventContext): Promise<FilterResponse> {
+        const shouldInclude = await this._callback(event, context);
```

Comment:
I see that it is awaited here, but I think that the contract of that callback should perhaps state that it's always a Promise<boolean>

Reply:
We don't need to 🙂